src/vanilla.ts

```diff
@@ -48,19 +60,17 @@ export type StateCreator<T extends State, CustomSetState = SetState<T>> = (
   api: StoreApi<T>
 ) => T
 
-export default function create<TState extends State>(
-  createState: StateCreator<TState>
-): StoreApi<TState> {
+export default function create<
+  TState extends State,
+  CustomSetState = SetState<TState>
+>(createState: StateCreator<TState, CustomSetState>): StoreApi<TState> {
   let state: TState
   const listeners: Set<StateListener<TState>> = new Set()
 
   const setState: SetState<TState> = (partial, replace) => {
     // TODO: Remove type assertion once https://github.com/microsoft/TypeScript/issues/37663 is resolved
     // https://github.com/microsoft/TypeScript/issues/37663#issuecomment-759728342
-    const nextState =
-      typeof partial === 'function'
-        ? (partial as (state: TState) => TState)(state)
-        : partial
+    const nextState = typeof partial === 'function' ? partial(state) : partial
```

Comment:
Hm, so we can remove the type assertion because of this PR, or was it already possible with the previous one?
Either way, let's remove L71-L72 comments.

Reply:
Ah, sorry my fix was sloppy it wasn't revert correctly but... it work okay. Sure, gonna remove comments.